File: ProductCatalog/src/main/java/com/Ecommerce/ProductCatalog/Product.java
package com.Ecommerce.ProductCatalog;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import javax.annotation.Generated;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "products")
public class Product {
    @Id
    private String id;

    private String unique_id;
    private String URL;
    private String name;
    private String category;
    private String description;
    private String brand;
    private int retailPrice;
    private int discountedPrice;
    private List<String> images;
    private int stock;
}

File: ApiGateway/Clients/Models/Product/Product.cs
using System;
using System.Collections.Generic;

namespace Clients.Models
{
    public class Product
    {
        public String id { get; set; }
        public String unique_id { get; set; }
        public String URL { get; set; }
        public String name { get; set; }
        public List<String> category { get; set; }
        public String description { get; set; }
        public String brand { get; set; }
        public int retailPrice { get; set; }
        public int discountedPrice { get; set; }
        public List<String> images { get; set; }
        public int stock { get; set; }
    }
}
